storage/remote/write_handler_test.go

```diff
@@ -412,23 +426,89 @@ func TestOutOfOrderSample_V2Message(t *testing.T) {
 			req.Header.Set("Content-Encoding", string(SnappyBlockCompression))
 			req.Header.Set(RemoteWriteVersionHeader, RemoteWriteVersion20HeaderValue)
 
-			appendable := &mockAppendable{latestSample: 100}
+			appendable := &mockAppendable{
+				commitErr:          tc.commitErr,
+				appendSampleErr:    tc.appendSampleErr,
+				appendHistogramErr: tc.appendHistogramErr,
+				appendExemplarErr:  tc.appendExemplarErr,
+				updateMetadataErr:  tc.updateMetadataErr,
+			}
 			handler := NewWriteHandler(log.NewNopLogger(), nil, appendable, []config.RemoteWriteProtoMsg{config.RemoteWriteProtoMsgV2})
 
 			recorder := httptest.NewRecorder()
 			handler.ServeHTTP(recorder, req)
 
 			resp := recorder.Result()
-			require.Equal(t, http.StatusBadRequest, resp.StatusCode)
+			require.Equal(t, tc.expectedCode, resp.StatusCode)
+			respBody, err := io.ReadAll(resp.Body)
+			require.NoError(t, err)
+			require.Equal(t, tc.expectedRespBody, string(respBody))
+
+			if tc.expectedCode == http.StatusInternalServerError {
+				// We don't expect writes for partial writes with retry-iable code.
+				expectHeaderValue(t, 0, resp.Header.Get("X-Prometheus-Remote-Write-Accepted-Samples"))
+				expectHeaderValue(t, 0, resp.Header.Get("X-Prometheus-Remote-Write-Accepted-Histograms"))
+				expectHeaderValue(t, 0, resp.Header.Get("X-Prometheus-Remote-Write-Accepted-Exemplars"))
+
+				require.Empty(t, len(appendable.samples))
+				require.Empty(t, len(appendable.histograms))
+				require.Empty(t, len(appendable.exemplars))
+				require.Empty(t, len(appendable.metadata))
+				return
+			}
+
+			// Double check mandatory 2.0 stats.
+			// writeV2RequestFixture has 2 series with 1 sample, 2 histograms, 1 exemplar each.
```

Comment:
None of the tests are adding additional accepted datapoints? they're all just adding additional data that will be rejected?

Reply:
Correct. Why are you asking? (:

Reply:
because it wasn't immediately clear, took me a couple minutes to understand this + the write request fixture creation